src/migration/MigrationExecutor.ts

```diff
@@ -55,17 +55,25 @@ export class MigrationExecutor {
         const pendingMigrations = allMigrations.filter(migration => {
             // check if we already have executed migration
             const executedMigration = executedMigrations.find(executedMigration => executedMigration.name === migration.name);
-            if (executedMigration)
-                return false;
 
-            // migration is new and not executed. now check if its timestamp is correct
-            if (lastTimeExecutedMigration && migration.timestamp < lastTimeExecutedMigration.timestamp)
-                throw new Error(`New migration found: ${migration.name}, however this migration's timestamp is not valid. Migration's timestamp should not be older then migrations already executed in the database.`);
-
-            // every check is passed means that migration was not run yet and we need to run it
-            return true;
+            return !executedMigration;
         });
 
+        // find migration with oldest timestamp
+        const oldestPendingMigrationTimestamp = Math.min(...pendingMigrations.map(migration => migration.timestamp));
+
+        // migration is new and not executed. now check if its timestamp is correct
+        if (lastTimeExecutedMigration && oldestPendingMigrationTimestamp < lastTimeExecutedMigration.timestamp) {
+            // find every executed migrations with a timestamp older than oldest pending migrations timestamp
+            for (let migration of executedMigrations) {
```

Comment:
why?
1. Find the oldest pending migration
2. Compare each executed migration with oldest pending migration and undo if it older then...
3. After concat with all migrations and sort